code/game/machinery/requests_console.dm

```diff
@@ -164,142 +165,128 @@ GLOBAL_LIST_EMPTY(allRequestConsoles)
-		print_label(ship_tag_name, ship_tag_index)
-		shipping_log += "<B>Shipping Label printed for [ship_tag_name]</b><br>[msgVerified]"
-		reset_message(1)
+	add_fingerprint(usr)
 
-	//Handle silencing the console
-	if(href_list["toggleSilent"])
-		silent = !silent
+	. = TRUE
+
+	switch(action)
+		if("writeInput")
+			if(reject_bad_text(params["write"]))
+				recipient = params["write"] //write contains the string of the receiving department's name
+
+				var/new_message = sanitize(input("Write your message:", "Awaiting Input", ""))
+				if(new_message)
+					message = new_message
+					screen = RCS_MESSAUTH
+					switch(params["priority"])
+						if("1")
+							priority = RQ_NORMALPRIORITY
+						if("2")
+							priority = RQ_HIGHPRIORITY
+						else
+							priority = RQ_NONEW_MESSAGES
+				else
+					reset_message(TRUE)
+
+		if("writeAnnouncement")
+			var/new_message = sanitize(input("Write your message:", "Awaiting Input", ""))
+			if(new_message)
+				message = new_message
+			else
+				reset_message(TRUE)
+
+		if("sendAnnouncement")
+			if(!announcementConsole)
+				return
+			announcement.Announce(message, msg_sanitized = TRUE)
+			reset_message(TRUE)
+
+		if("department")
+			if(!message)
+				return
+			var/log_msg = message
+			var/pass = FALSE
+			screen = RCS_SENTFAIL
+			for(var/obj/machinery/message_server/MS in GLOB.machines)
+				if(!MS.active)
+					continue
+				MS.send_rc_message(ckey(params["department"]), department, log_msg, msgStamped, msgVerified, priority)
+				pass = TRUE
+			if(pass)
+				screen = RCS_SENTPASS
+				if(recipient in ENGI_ROLES)
+					radiochannel = "Engineering"
+				if(recipient in SEC_ROLES)
+					radiochannel = "Security"
+				if(recipient in MISC_ROLES)
+					radiochannel = "Service"
+				if(recipient in MED_ROLES)
+					radiochannel = "Medical"
+				if(recipient in COM_ROLES)
+					radiochannel = "Command"
+				if(recipient in SCI_ROLES)
+					radiochannel = "Science"
+				if(recipient == "AI")
+					radiochannel = "AI Private"
+				if(recipient == "Cargo Bay")
+					radiochannel = "Supply"
+				message_log += "Message sent to [recipient] at [station_time_timestamp()] - [message]"
+				Radio.autosay("Alert; a new requests console message received for [recipient] from [department]", null, "[radiochannel]")
+			else
+				atom_say("No server detected!")
+
+		//Handle screen switching
+		if("setScreen")
+			var/tempScreen = text2num(params["setScreen"])
+			if(tempScreen == RCS_ANNOUNCE && !announcementConsole)
+				return
+			if(tempScreen == RCS_VIEWMSGS)
+				for(var/obj/machinery/requests_console/Console in GLOB.allRequestConsoles)
+					if(Console.department == department)
+						Console.newmessagepriority = RQ_NONEW_MESSAGES
+						Console.icon_state = "req_comp0"
+						Console.set_light(1)
+			if(tempScreen == RCS_MAINMENU)
+				reset_message()
+			screen = tempScreen
+
+		if("shipSelect")
+			ship_tag_name = params["shipSelect"]
+			ship_tag_index = GLOB.TAGGERLOCATIONS.Find(ship_tag_name)
```

Comment:
Needs a check to whether the shipping destination is valid.

Reply:
```dm
			if(!ship_tag_index)
				error_message = "Please select a destination."
```